src/v/config/configuration.cc

```diff
@@ -3275,7 +3275,18 @@ configuration::configuration()
       "internal-only configuration and should be enabled only after consulting "
       "with Redpanda Support or engineers.",
       {.needs_restart = needs_restart::yes, .visibility = visibility::user},
-      false) {}
+      false)
+  , tls_min_version(
+      *this,
+      "tls_min_version",
+      "The minimum version of TLS that Redpanda will support.",
+      {.needs_restart = needs_restart::yes, .visibility = visibility::user},
+      tls_version::v1_2,
+      {tls_version::v1_0,
+       tls_version::v1_1,
+       tls_version::v1_2,
+       tls_version::v1_3},
+      legacy_default<tls_version>(tls_version::v1_0, legacy_version{12})) {}
```

Comment:
So [this](https://github.com/redpanda-data/redpanda/issues/15674) is the issue from late last year that @dotnwat was referring to. I think we're subject to this weird behavior here as well. You might try:

- upgrade from `<= logical 12` to HEAD
- set the min tls version to 1.2
- restart the broker
- see whether the min tls version is set back to 1.0

In this case I think the escape hatch would be to set something other than the current version non-legacy default (e.g. 1.1 or 1.3), but it's worth being aware of, esp. if this is a compliance concern.

Reply:
Thanks for flagging.  I'll write up a DT test for this specifically

Reply:
Yeah thanks for flagging that is an issue.  Setting to v1.2 and then restarting resets it back to v1.0....